python/lsst/pipe/tasks/extended_psf.py

```diff
@@ -239,3 +245,250 @@ def readFits(cls, filename):
+    )
+    do_mag_cut = pexConfig.Field(
+        dtype=bool,
+        doc="Apply magnitude cut before stacking?",
+        default=False
+    )
+    mag_limit = pexConfig.Field(
+        dtype=float,
+        doc="Magnitude limit, in Gaia G; all stars brighter than this value will be stacked",
+        default=18
+    )
+
+
+class StackBrightStarsTask(pipeBase.CmdLineTask):
+    """Stack bright stars together to build an extended PSF model.
+    """
+    ConfigClass = StackBrightStarsConfig
+    _DefaultName = "stack_bright_stars"
+
+    def __init__(self, initInputs=None, *args, **kwargs):
+        pipeBase.CmdLineTask.__init__(self, *args, **kwargs)
+
+    def _set_up_stacking(self, example_stamp):
+        """Configure stacking statistic and control from config fields.
+        """
+        stats_control = afwMath.StatisticsControl()
+        stats_control.setNumSigmaClip(self.config.num_sigma_clip)
+        stats_control.setNumIter(self.config.num_iter)
+        bad_masks = self.config.bad_mask_planes
+        if bad_masks:
+            and_mask = example_stamp.mask.getPlaneBitMask(bad_masks[0])
+            for bm in bad_masks[1:]:
+                and_mask = and_mask | example_stamp.mask.getPlaneBitMask(bm)
+            stats_control.setAndMask(and_mask)
+        stats_flags = afwMath.stringToStatisticsProperty(self.config.stacking_statistic)
+        return stats_control, stats_flags
+
+    def run(self, bss_ref_list, region_name=None):
+        """Read input bright star stamps and stack them together.
+
+        The stacking is done iteratively over smaller areas of the final model
+        image to allow for a great number of bright star stamps to be used.
+
+        Parameters
+        ----------
+        bss_ref_list : `list` of
+                `lsst.daf.butler._deferredDatasetHandle.DeferredDatasetHandle`
+            List of available bright star stamps data references.
+        region_name : `str`, optional
+            Name of the focal plane region, if applicable. Only used for
+            logging purposes, when running over multiple such regions
+            (typically from `MeasureExtendedPsfTask`)
+        """
+        log_message = f'Building extended PSF from stamps extracted from {len(bss_ref_list)} detector images '
+        if region_name:
+            log_message += f'for region "{region_name}".'
+        self.log.info(log_message)
+        # read in example set of full stamps
+        for bss_ref in bss_ref_list:
+            if not isinstance(bss_ref, DeferredDatasetHandle):
+                if not bss_ref.datasetExists("brightStarStamps"):
+                    self.log.warn("Could not find %s %s; skipping it", "brightStarStamps", bss_ref.dataId)
+                    bss_ref_list.remove(bss_ref)
+                    continue
+            bss = bss_ref.get(datasetType="brightStarStamps", immediate=True)
+            break
+        example_stamp = bss[0].stamp_im
```

Comment:
From reading this block, it seems like the purpose is to generate `example_stamp`. I was a little confused as to why two `if` statements were required, however, after having checked with you offline, it appears that it may be possible to inject a valid `DeferredDatasetHandle` that does not contain any `brightStarStamps` dataset types. In such a case however, L350 would still raise an exception and crash.

It might be cleaner to replace the two `if` statements with a try/except block here, i.e.:
```
for bss_ref in bss_ref_list:
    try:
        bss = bss_ref.get(datasetType="brightStarStamps", immediate=True)
        example_stamp = bss[0].stamp_im
    except ...
```

Reply:
Mystery solved!

This is just vestigial code left from when I had a gen2 version of this Task. The first `if` clause was only ever `True` when the inputs to `run` were those given by the gen2 `runDataRef`. 
I should have removed all this (and the code in lines 367-371)  when I got rid of the gen2 version. Apologies!

Note this also (sort of) explains the weird syntax I went with; I was probably cargoculting what is done in many other places in the stack when we want stuff to be able to run on both gen2 and gen3 (eg [here](https://github.com/lsst/pipe_tasks/blob/387f8f07a2b66205f9fa6bda9a89dcdbbef3f64c/python/lsst/pipe/tasks/dcrAssembleCoadd.py#L542-L547)), where the first `if` is really just the beginning of the handle-gen2 code block. Except in my case, I just had the one check, so it really could have been a single `if` statement (in fact I used  to deal with this [somewhat differently](https://github.com/lsst/pipe_tasks/blob/b64fabfa9b71ed0b1609c282480c22d7da685a86/python/lsst/pipe/tasks/measureExtendedPsf.py#L142-L145), and closer to what you suggested here, which is probably why I wrote things this way before switching to the syntax we see here).

Anyway - I've removed it all now, as I should have from the get go. Sorry again.

The good thing is this pushed me into testing a bunch of things, including rerunning on dataIds where I used to have trouble in gen2, and can confirm it does not occur in gen3. It also made me realize I had to file and fix [DM-29862](https://jira.lsstcorp.org/browse/DM-29862)


TL;DR: this code should have never made it to the PR (it's how I used to solve a gen2-specific problem)